docs/pages/0058-advanced-configuration.rst
.. _Production:

Advanced Configuration Topics
=============================

.. contents:: :local:

.. include:: /pages/advanced-topics/0010-introduction.rst
.. include:: /pages/advanced-topics/0020-custom-plug-ins.rst
.. include:: /pages/advanced-topics/0030-quantum-HA.rst
.. include:: /pages/advanced-topics/0040-bonding.rst


docs/pages/installing-configuring-cobbler-2.rst
Installation Instructions
=========================

.. contents:: :local:


Installing & Configuring Cobbler
--------------------------------

Cobbler is a bare metal provisioning system which performs bare metal provisioning and initial installation of Linux on OpenStack nodes. Luckily, we already have a Puppet master installed, so we can install Cobbler using Puppet in a few seconds instead of doing it manually.

Using Puppet to install Cobbler
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

On Puppet master:

* ``vi /etc/puppet/manifests/site.pp``

* Copy the content of "fuel/deployment/puppet/cobbler/examples/site.pp" into "/etc/puppet/manifests/site.pp":
    .. literalinclude:: ../../deployment/puppet/cobbler/examples/site_fordocs.pp

* Make the following changes in that file:
    * Replace IP addresses and ranges according to your network setup. Replace "your-domain-name.com" with your domain name.
    * Uncomment the required OS distributions. They will be downloaded and imported into Cobbler during Cobbler installation.
    * Change the location of ISO image files to either a local mirror or the fastest available Internet mirror.

* Once the configuration is there, Puppet will know that Cobbler must be installed on the fuel-pm machine. Once Cobbler is installed, the right distro and profile will be automatically added to it. OS image will be downloaded from the mirror and put into Cobbler as well.

* It is necessary to note that in the proposed network configuration the snippet above includes Puppet commands to configure forwarding on Cobbler node to make external resources available via the 10.0.0.0/24 network which is used during the installation process (see "enable_nat_all" and "enable_nat_filter")

* run puppet agent to actually install Cobbler on fuel-pm
    * ``puppet agent --test``

Testing cobbler
~~~~~~~~~~~~~~~

* you can check that Cobbler is installed successfully by opening the following URL from your host machine:
    * http://fuel-pm/cobbler_web/ (u: cobbler, p: cobbler)
* now you have a fully working instance of Cobbler. Moreover, it is fully configured and capable of installing the chosen OS (CentOS 6.3, RHEL 6.3, or Ubuntu 12.04) on the target OpenStack nodes


CHANGELOG
3.0-alpha-174-g94a98d6
- Merge from master branch
- Rsyslog tuning
- Puppet debug output
- Centos 6.4

2.1-folsom-docs-324-g61d1599
- Grizzly support for centos simple
- Option for PKI auth for keystone (grizzly native)
- Nova-conductor as generic nova service at compute nodes
- CI scripts changes for grizzly tempest (host only routed IP addresses for public pool)
- 


fuel_test/settings.py
import os

OS_FAMILY = os.environ.get('OS_FAMILY', "centos")
PUPPET_GEN = os.environ.get('PUPPET_GEN', "2")

DEFAULT_IMAGES = {
    'centos': '/var/lib/libvirt/images/centos63-cobbler-base.qcow2',
    'ubuntu': '/var/lib/libvirt/images/ubuntu-12.04.1-server-amd64-base.qcow2',
}

BASE_IMAGE = os.environ.get('BASE_IMAGE', DEFAULT_IMAGES.get(OS_FAMILY))

PUPPET_VERSIONS = {
    'centos': {
        "2": '2.7.19-1.el6',
        "3": '3.0.1-1.el6',
        },
    'ubuntu': {
        "2": '2.7.19-1puppetlabs2',
        "3": '3.0.1-1puppetlabs1'
    },
}

PUPPET_VERSION = PUPPET_VERSIONS.get(OS_FAMILY).get(PUPPET_GEN)
PUPPET_MASTER_VERSION = PUPPET_VERSIONS.get('centos').get(PUPPET_GEN)

PUPPET_CLIENT_PACKAGES = {
    'centos': {
        "2": 'puppet-2.7.19-1.el6',
        "3": 'puppet-3.0.1-1.el6',
        },
    'ubuntu': {
        "2": 'puppet=2.7.19-1puppetlabs2 puppet-common=2.7.19-1puppetlabs2',
        "3": 'puppet=3.0.1-1puppetlabs1 puppet-common=3.0.1-1puppetlabs1'
    },
}

PUPPET_CLIENT_PACKAGE = PUPPET_CLIENT_PACKAGES.get(OS_FAMILY).get(PUPPET_GEN)

ERROR_PREFIXES = {
    "2": "err: ",
    "3": "Error: ",
}

ERROR_PREFIX = ERROR_PREFIXES.get(PUPPET_GEN)

WARNING_PREFIXES = {
    "2": "warning: ",
    "3": "Warning: ",
}

WARNING_PREFIX = WARNING_PREFIXES.get(PUPPET_GEN)

PUPPET_MASTER_SERVICE = 'thin'

ADMIN_USERNAME = 'admin'
ADMIN_PASSWORD = 'nova'
ADMIN_TENANT_ESSEX = 'openstack'
ADMIN_TENANT_FOLSOM = 'admin'

CIRROS_IMAGE = os.environ.get('CIRROS_IMAGE', 'http://srv08-srt.srt.mirantis.net/cirros-0.3.0-x86_64-disk.img')
CONTROLLERS = int(os.environ.get('CONTROLLERS', 3))
COMPUTES = int(os.environ.get('COMPUTES', 3))
STORAGES = int(os.environ.get('STORAGES', 0))
PROXIES = int(os.environ.get('PROXIES', 0))
QUANTUMS = int(os.environ.get('QUANTUMS', 0))
DEFAULT_RAM_SIZE = int(os.environ.get('DEFAULT_RAM_SIZE', 1024))
COMPUTE_RAM_SIZE = int(os.environ.get('COMPUTE_RAM_SIZE', 2048))

EMPTY_SNAPSHOT = os.environ.get('EMPTY_SNAPSHOT', 'empty')
OPENSTACK_SNAPSHOT = os.environ.get('OPENSTACK_SNAPSHOT', 'openstack')

INTERFACE_ORDER = ('public', 'internal', 'private')

INTERFACES = {
    'public': 'eth0',
    'internal': 'eth1',
    'private': 'eth2',
}

FORWARDING = {
    'public': os.environ.get('PUBLIC_FORWARD', 'nat'),
    'internal': None,
    'private': None,
}

DHCP = {
    'public': True,
    'internal': False,
    'private': False,
}

DEFAULT_POOLS = {
    'centos': {
        'public': '10.108.0.0/16:24',
        'private': '10.108.0.0/16:24',
        'internal': '10.108.0.0/16:24',
        },
    'ubuntu': {
        'public': '10.107.0.0/16:24',
        'private': '10.107.0.0/16:24',
        'internal': '10.107.0.0/16:24',
        },
}

POOLS = {
    'public': os.environ.get('PUBLIC_POOL',
                             DEFAULT_POOLS.get(OS_FAMILY).get('public')).split(':'),
    'private': os.environ.get('PRIVATE_POOL',
                              DEFAULT_POOLS.get(OS_FAMILY).get('private')).split(':'),
    'internal': os.environ.get('INTERNAL_POOL',
                               DEFAULT_POOLS.get(OS_FAMILY).get('internal')).split(':')
}

TEST_REPO = os.environ.get('TEST_REPO', 'false') == 'true'
EXIST_TAR = os.environ.get('EXIST_TAR', None)
CREATE_SNAPSHOTS = os.environ.get('CREATE_SNAPSHOTS', 'true') == 'true'
CLEAN = os.environ.get('CLEAN', 'true') == 'true'
ISO_IMAGE = os.environ.get('ISO_IMAGE', '~/fuel-centos-6.4-x86_64-3.0.iso')
USE_ISO = os.environ.get('USE_ISO', 'true') == 'true'
PARENT_PROXY = os.environ.get('PARENT_PROXY', '')
PROFILES_COBBLER_COMMON = {
    'centos': 'centos64_x86_64',
    'ubuntu': 'ubuntu_1204_x86_64'
}

CURRENT_PROFILE = PROFILES_COBBLER_COMMON.get(OS_FAMILY)

ASTUTE_USE = os.environ.get('ASTUTE_USE', 'true') == 'true'
DOMAIN_NAME = os.environ.get('DOMAIN_NAME', '.localdomain')
PUPPET_AGENT_COMMAND = 'puppet agent -tvd --evaltrace 2>&1'
SETUP_TIMEOUT = int(os.environ.get('SETUP_TIMEOUT', 600))

